src/bsoncxx/array/view.cpp

```diff
@@ -47,19 +47,19 @@ view::iterator& view::iterator::operator++() {
     }
 
     bson_iter_t i;
-    i.raw = _element.raw;
-    i.len = _element.length;
-    i.next_off = _element.offset;
+    i.raw = _element.raw();
+    i.len = _element.length();
+    i.next_off = _element.offset();
     bson_iter_next(&i);
 
     if (!bson_iter_next(&i)) {
-        _element.raw = nullptr;
-        _element.length = 0;
-        _element.offset = 0;
+        _element.raw(nullptr);
```

Comment:
I don't know why it was written this way. Perhaps it's slightly more efficient? But I'm happy to change it.